### PR TITLE
more explicitly explain the UB around immutable extern statics

### PR DESCRIPTION
**Pull request by RalfJung**
Fixes https://github.com/rust-lang/unsafe-code-guidelines/issues/514

### CODE REVIEW

File: src/items/external-blocks.md [hunk 1]
```diff
@@ -56,6 +56,8 @@ of initializing the static.
 Extern statics can be either immutable or mutable just like [statics] outside of external blocks.
 An immutable static *must* be initialized before any Rust code is executed. It is not enough for
 the static to be initialized before Rust code reads from it.
+One Rust code runs, mutating an immutable static (from inside or outside Rust) is UB,
```
Comment (bjorn3) [inline src/items/external-blocks.md:59]: *Once

**Review by nikomatsakis — DISMISSED** [review]
Clarification requested

**Review by tmandry — APPROVED** [review]

File: src/items/external-blocks.md [hunk 2]
```diff
@@ -56,6 +56,8 @@ of initializing the static.
 Extern statics can be either immutable or mutable just like [statics] outside of external blocks.
 An immutable static *must* be initialized before any Rust code is executed. It is not enough for
 the static to be initialized before Rust code reads from it.
+Once Rust code runs, mutating an immutable static (from inside or outside Rust) is UB,
+except if the mutation happens inside an `UnsafeCell`.
```
[the remark this answers is not in the document]
Comment (RalfJung) [reply src/items/external-blocks.md:60]: Yes.

[the remark this answers is not in the document]
Comment (traviscross) [reply src/items/external-blocks.md]: I've pushed a commit to add that clarification to the text.

**Comment by traviscross** [comment]
@rustbot labels -I-lang-nominated

We discussed this in the lang triage call today.  Other than a clarification answered and now included in the text, we were OK with this.

**Pull request merged**